resources/views/classes/create.blade.php

```diff
@@ -1,67 +1,440 @@
+					<input class="is-checkradio" type="checkbox" name="AllowIndividualDayRegistration" id="AllowIndividualDayRegistration" value="0" v-model="form.AllowIndividualDayRegistration" >
+					<label for="AllowIndividualDayRegistration"></label>
+				</div>
+			</div>
+		</div>
+		<div class="column">
+			<div class="field">
+				<label class="label">Password</label>
+				<div class="control">
+					<input class="input" type="text" placeholder="Online registration will require this password" v-model="form.Password" name="Password">
+					<span class="help is-danger" v-if="form.errors.has('Password')" v-text="form.errors.get('Password')"></span>
+				</div>
+			</div>
+		</div>
+		<div class="column">
+			<div class="field">
+				<label class="label">Class Charge</label>
+				<div class="field is-expanded">
+      		<div class="field has-addons">
+        		<p class="control">
+          		<a class="button is-static">$</a>
+        		</p>
+						<p class="control is-expanded">
+							<input class="input" type="text" placeholder="This will override any applicible automatic rate." v-model="form.ClassCharge" name="ClassCharge">
+						</p>
+      		</div>
+    		</div>
+			</div>
+		</div>
+	</div>
+	<div class="columns is-centered">
+		<div class="column is-narrow">
+			<button class="button is-primary" :disabled="form.errors.any()" type="submit">Save</button></button>
+		</div>
+	</div>
+</div>
+
+	<script>
+
+		class Errors {
+
+			constructor() {
+				this.errors = {};
+			}
+
+			has(field) {
+				return this.errors.hasOwnProperty(field);
+			}
+
+			any() {
+				return Object.keys(this.errors).length > 0;
+			}
+
+			get(field) {
+				if (this.errors[field]) {
+					return this.errors[field][0];
+				}
+			}
+
+			record(errors) {
+				this.errors = errors;
+			}
+
+			clear(field) {
+				if (field) {
+					delete this.errors[field];
+					return;
+				}
+				this.errors = {};
+			}
+		}
+
+		class Form {
+
+			constructor(data) {
+				this.originalData = data;
+				for (let field in data) {
+					this[field] = data[field];
+				}
+				this.errors = new Errors();
+			}
+
+			data() {
+				let data = {};
+				for (let property in this.originalData) {
+					data[property] = this[property];
+				}
+				return data;
+			}
+
+			reset() {
+				for (let field in originalData) {
+					this[field] = '';
+				}
+				this.errors.clear();
+			}
+
+			submit (requestType, url) {
+				return new Promise((resolve, reject) => {
+					axios[requestType](url, this.data())
+							.then(response => {
+								this.onSuccess(response.data);
+								resolve(response.data);
+							})
+							.catch(error => {
+								this.onFail(error.response.data.errors);
+								reject(error.response.data.errors);
+							})
+				});
+			}
+
+			onSuccess(response) {
+				alert(response.data.message);
+				this.reset();
+			}
+
+			onFail(errors) {
+				this.errors.record(errors);
+			}
+		}
+
+		var classCreate = new Vue({
+			el: '#app',
+
+			data: {
+				form: new Form({
```

Comment:
YIKES

Reply:
Ok, so I kind of skipped the steps of moving the errors and form classes to its own file. My bad.